app/models/notification_type.rb

```diff
@@ -0,0 +1,34 @@
+class NotificationType < ActiveRecord::Base
+  AUDIENCE_USER     = 'user'.freeze
+  AUDIENCE_GROUP    = 'usergroup'.freeze
+  AUDIENCE_TAXONOMY = 'taxonomy'.freeze
+  AUDIENCE_GLOBAL   = 'global'.freeze
+  AUDIENCE_ADMIN    = 'admin'.freeze
+  has_many :notifications
+  validates :message, :presence => true
+  validates :level, :inclusion => { :in => %w(success error warning info) }, :presence => true
+  validates :audience, :inclusion => { :in => [AUDIENCE_USER, AUDIENCE_GROUP, AUDIENCE_TAXONOMY, AUDIENCE_GLOBAL, AUDIENCE_ADMIN] }, :presence => true
+  validates :expires_in, :numericality => {:greater_than => 0}
+  before_validation :set_default_expiry
+
+  def subscriber_ids(initiator, subject = nil)
+    case audience
+    when AUDIENCE_GLOBAL
+      User.reorder('').uniq.pluck(:id)
```

Comment:
reorder is just faster (no need to sort by default scope - now or in the future), you are probably right about the uniq.